helix-core/src/main/java/org/apache/helix/task/TaskStateModel.java

```diff
@@ -283,6 +289,36 @@ public void reset() {
     }
   }
 
+  /**
+   * Loads Task and TaskFactory classes for command input
+   */
+  private boolean loadNewTask(String command) {
+    try {
+      // Read ZNRecord containing task definition information.
+      ZNRecord taskConfig = _manager.getHelixDataAccessor().getBaseDataAccessor()
```

Comment:
If this path doesn't exist, does it throw exception here or return null? If exception, do we need to handle the exception? If null, we need to do a check in the next line of code to prevent NPE.

Reply:
Thank you, Molly. Good catch! I think I should add the appropriate option to throw exception and then handle it.
```suggestion
      ZNRecord taskConfig = _manager.getHelixDataAccessor().getBaseDataAccessor()
          .get("/" + _manager.getClusterName() + "/TASK_DEFINITION",
              null, AccessOption.THROW_EXCEPTION_IFNOTEXIST);
```